include/ncengine/graphics/SkeletalAnimator.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "ncengine/ecs/Component.h"
+#include "ncengine/graphics/SkeletalAnimationTypes.h"
+
+#include <memory>
+
+namespace nc::graphics
+{
+class SkeletalAnimator : public ComponentBase
+{
+    public:
+        SkeletalAnimator(Entity entity, std::string assetUid);
+        SkeletalAnimator(SkeletalAnimator&& other) noexcept;
+        SkeletalAnimator(SkeletalAnimator& other) = delete;
+        auto operator=(SkeletalAnimator&& other) noexcept -> SkeletalAnimator&;
+        auto operator=(SkeletalAnimator& other) -> SkeletalAnimator& = delete;
+        ~SkeletalAnimator() noexcept;
+
+        // Main
```

Comment:
Use doxygen-style comments for public facing stuff. I'm ok if you want to exclude them for now.